### ISSUE TITLE
EPIC: Plan for next steps

### ISSUE THREAD

**Issue by xmnlab** [osl-incubator/conda-forge-warning#2]
from @jaimergp, in the conda-forge element channel:

* [ ] #3
* [ ] #4
* [ ] #5
* [ ] #6
* [ ] #7
* [ ] #8
* [ ] #9

https://github.com/osl-incubator/conda-forge-warning/issues/2#issuecomment-1798279218
Some more suggestions by @beckermr:

- [ ] #10
- [ ] #11
- [ ] #12

**Comment by jaimergp** [comment]
Some more suggestions by @beckermr:

- [ ] repos that have tons of downstream children have only a few maintainers
- [ ] repos whose build times are close to 6 hours
- [ ] repos with failed builds on main

**Comment by xmnlab** [comment]
@jaimergp and @beckermr thank you for your suggestions! Appreciate that!

**Comment by h-vetinari** [comment]
> Feedstocks where the autotickbot has accummulated more than 3 open PRs

It might also make sense to differentiate between version updates and migrations. If there's no open version update PRs, that might be just because it's a package with very low release cadence, or because the bot failed (see below). Alternatively, if many failed migration PRs pile up, then the feedstock is almost always dormant/dead.

Speaking about the version bot failures, there's a [section](https://conda-forge.org/status/#version_updates) on the conda-forge status page dealing with feedstocks where the autotickbot failed to open a PR in the first place - i.e. one step before you can reasonably rely on that metric for most feedstocks.

We [discussed](https://github.com/regro/cf-scripts/issues/1341) a long time ago to make automated issues for these, but that hasn't materialized yet. If it's easier to put that information into a dashboard, that might already be a substantial improvement.

Other than that:
* This effort probably also needs a whitelist. Things like `conda-forge-{pinning,repodata-patches}-feedstock` will never go out of date, and pytorch/tensorflow are under constant attention as well (only very hard to build, hence the pent up PRs).
* It should ideally pick up feedstocks from https://github.com/conda-forge/conda-forge.github.io/issues/1724 / https://github.com/conda-forge/conda-forge.github.io/issues/1861, and avoid high-volume (not least due to many `abi_migration_branches`) feedstocks like [arrow](https://github.com/conda-forge/arrow-cpp-feedstock)
  * for that, something like "last-non-bot-commit" or "last-non-bot-comment" would probably be helpful metrics
* discount open PRs that are in draft mode - sometimes larger refactors on a feedstock are open for a long time, but that's not a sign of bad health
* don't weight issues too heavily; often there are questions that are left open for reasons of visibility, or problems that aren't fixable in the foreseeable future, or a list of todos for further incremental improvements etc.
* Some sensitivity about nomenclature. Issuing a "warning" about feedstocks with 1 open issue and one open PR is not going to be helpful IMO, especially in an age where automated tools are liable to start screaming at their users about these things (c.f. the situation with CVEs).
  * I consider it very important that this does not become a blunt hammer used against still-active maintainers, _especially_ where it's a false positive.
  * The simplest is probably either a cut-off below which nothing is displayed, or a category below "Warning" that just lists the scraped status of these feedstocks under a more neutral heading.

**Comment by xmnlab** [comment]
Thanks @h-vetinari for your inputs!
I will take a time to process all the inputs here.
